pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go

```diff
@@ -253,3 +266,60 @@ func TestPartitionedTenantStreamingEndToEnd(t *testing.T) {
 		testTenantStreaming(t, false /* withInitialScan */)
 	})
 }
+
+func TestTenantStreamingProducerJobTimedOut(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+
+	c, cleanup := createTenantStreamingClusters(ctx, t, tenantStreamingClustersArgs{
+		srcTenantID: roachpb.MakeTenantID(10),
+		srcInitFunc: func(t *testing.T, sysSQL *sqlutils.SQLRunner, tenantSQL *sqlutils.SQLRunner) {
+			sysSQL.ExecMultiple(t, clusterSettingStatements(srcClusterSetting)...)
+			tenantSQL.Exec(t, `
+	CREATE DATABASE d;
+	CREATE TABLE d.t1(i int primary key, a string, b string);
+	CREATE TABLE d.t2(i int primary key);
+	INSERT INTO d.t1 (i) VALUES (42);
+	INSERT INTO d.t2 VALUES (2);
+	UPDATE d.t1 SET b = 'world' WHERE i = 42;
+	`)
+		},
+		srcNumNodes:  1,
+		destTenantID: roachpb.MakeTenantID(20),
+		destInitFunc: func(t *testing.T, sysSQL *sqlutils.SQLRunner, tenantSQL *sqlutils.SQLRunner) {
+			sysSQL.ExecMultiple(t, clusterSettingStatements(destClusterSetting)...)
+		},
+		destNumNodes: 1,
+	})
+	defer cleanup()
+
+	var startTime string
+	c.srcSysSQL.QueryRow(t, "SELECT cluster_logical_timestamp()").Scan(&startTime)
+	// initial scan
+	producerJobID, ingestionJobID := c.startStreamReplication("" /* startTime */)
+
+	jobutils.WaitForJobToRun(c.t, c.srcSysSQL, jobspb.JobID(producerJobID))
+	jobutils.WaitForJobToRun(c.t, c.destSysSQL, jobspb.JobID(ingestionJobID))
+
+	// Make producer job easily times out
+	c.srcSysSQL.ExecMultiple(t, clusterSettingStatements(map[string]string{
+		`stream_replication.job_liveness_timeout`: `'100ms'`,
+	})...)
+	log.Info(ctx, "changed timeout")
```

Comment:
If we think this is a useful log message, consider using `t.Log` instead of log.Info here.